<launch>
    
<arg name="markerSize"      default="0.1"/>    <!-- in m -->
  <arg name="ref_frame"       default="base_link"/>  <!-- leave empty and the pose will be published wrt param parent_name -->

  <!-- Run the map server -->
   
  <include file="$(find ucar_controller)/launch/base_driver.launch" > </include>

  <include file="$(find ydlidar)/launch/ydlidar.launch" > </include>

  <node name="map_server" pkg="map_server" type="map_server" args="$(find ucar_nav)/maps/map_xunfei1.yaml" output="screen">
   <param name="frame_id" value="map" />
  </node> 

<!-- <param name="base_global_planner" value="voronoi_planner/VoronoiPlanner"/>  -->

  <include file="$(find ucar_nav)/launch/config/amcl/amcl_omni.launch" > </include>
  <!-- <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_global_planner" value="global_planner/GlobalPlanner"/>
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/>
    <rosparam file="$(find ucar_nav)/launch/dwa/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find ucar_nav)/launch/dwa/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find ucar_nav)/launch/dwa/base_global_planner_params.yaml" command="load" />
    <rosparam file="$(find ucar_nav)/launch/dwa/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find ucar_nav)/launch/dwa/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find ucar_nav)/launch/dwa/global_costmap_params.yaml" command="load" />  -->

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" clear_params="true">
    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
    <!-- <param name="controller_frequency" value="20.0" /> -->
    <!-- <param name="controller_patience" value="20.0" /> -->
    <rosparam file="$(find ucar_nav)/launch/config/teb/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find ucar_nav)/launch/config/teb/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find ucar_nav)/launch/config/teb/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find ucar_nav)/launch/config/teb/global_costmap_params.yaml" command="load" />

    <rosparam file="$(find ucar_nav)/launch/config/teb/move_base_params.yaml" command="load" />
    <rosparam file="$(find ucar_nav)/launch/config/teb/base_global_planner_params.yaml" command="load" />
    <rosparam file="$(find ucar_nav)/launch/config/teb/teb_local_planner_params.yaml" command="load" />
    <!-- <remap from="/cmd_vel" to="/vesc/cmd_vel" /> -->
  </node> 
<!--usb_cam-->

<!--marker-->

    <!-- <node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher" output="screen">
        <remap from="/camera_info" to="/usb_cam/camera_info" />
        <remap from="/image" to="/usb_cam/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="reference_frame"    value="$camera_frame"/>  
        <param name="camera_frame"       value="$camera_frame"/>
    </node> -->
 
<!--*****************others******************-->
<include file="$(find xf_mic_asr_offline)/launch/xf_mic_asr_offline.launch" />

<!-- <include file="$(find rf2o_laser_odometry)/launch/rf2o_laser_odometry.launch" /> -->


<!--  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node>-->

  <!-- <node name="curvature_cal" pkg="ucar_nav" type="curvature_cal.py" output="screen"/> -->
  <!-- <rosparam command="load" file="$(find ucar_nav)/launch/config/move_base/my_navigation.yaml"/> -->

</launch>
